src/cplscheme/tests/SerialImplicitCouplingSchemeTest.cpp

```diff
@@ -1265,20 +1536,32 @@ BOOST_AUTO_TEST_CASE(testInitializeData)
   cplScheme.initialize(0.0, 1);
 
   if (context.isNamed(nameParticipant0)) {
-    BOOST_TEST(testing::equals(receiveCouplingData->values(), Eigen::Vector3d(0.0, 0.0, 0.0)));
+    // ensure that read data is uninitialized
     BOOST_TEST(receiveCouplingData->values().size() == 3);
+    BOOST_TEST(testing::equals(receiveCouplingData->values(), Eigen::Vector3d(0.0, 0.0, 0.0)));
     BOOST_TEST(receiveCouplingData->previousIteration().size() == 3);
-    BOOST_TEST(testing::equals(sendCouplingData->values()(0), 0.0));
+    BOOST_TEST(testing::equals(receiveCouplingData->previousIteration(), Eigen::Vector3d(0.0, 0.0, 0.0)));
+    // ensure that write data is uninitialized
     BOOST_TEST(sendCouplingData->values().size() == 1);
+    BOOST_TEST(testing::equals(sendCouplingData->values()(0), 0.0));
     BOOST_TEST(sendCouplingData->previousIteration().size() == 1);
+    BOOST_TEST(testing::equals(sendCouplingData->previousIteration()(0), 0.0));
+
     BOOST_TEST(fixture.isImplicitCouplingScheme(cplScheme));
     cplScheme.initializeData();
     BOOST_TEST(cplScheme.hasDataBeenReceived());
+    // ensure that initial data was read
+    BOOST_TEST(receiveCouplingData->values().size() == 3);
     BOOST_TEST(testing::equals(receiveCouplingData->values(), Eigen::Vector3d(1.0, 2.0, 3.0)));
     BOOST_TEST(receiveCouplingData->previousIteration().size() == 3);
-    BOOST_TEST(testing::equals(receiveCouplingData->previousIteration(), Eigen::Vector3d(0.0, 0.0, 0.0)));
-    BOOST_TEST(sendCouplingData->previousIteration().size() == 1);
+    BOOST_TEST(testing::equals(receiveCouplingData->previousIteration(), Eigen::Vector3d(1.0, 2.0, 3.0)));
```

Comment:
I think this here is the important part: If the initial data should be considered for extrapolation, we also have to initialize the `previousIteration` with the initial data. Otherwise the data is lost and the extrapolation cannot use it. I will try to prepare a fix now. Maybe the fix will also end up somewhere else (not in `previousIteration`, but somewhere in `Waveform`)